### PR TITLE
[HACK Week] Improve Support Request form error and identity handling

### PR DESCRIPTION
**Pull request by ThomazFB**
Summary
==========
Refactor how the `ZendeskTicketRepository` reacts when there's no Contact identity information set by creating an error type system that is verified inside the `SupportRequestFormViewModel`, making sure that we will never attempt to submit the request without a valid identity set.

How to Test
==========
1. Inside the `ZendeskSettings` class, hard code the `isIdentitySet` property to `false` and make sure there's absolute no way possible to submit the request.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/push/NotificationMessageHandler.kt [hunk 1]
```diff
@@ -98,20 +89,6 @@ class NotificationMessageHandler @Inject constructor(
...
-            val zendeskNote = NotificationModel(
-                noteId = ZENDESK_PUSH_NOTIFICATION_ID,
-                remoteNoteId = ZENDESK_PUSH_NOTIFICATION_ID.toLong()
-            ).toAppModel(resourceProvider)
-            notificationBuilder.buildAndDisplayZendeskNotification(
-                channelId = resourceProvider.getString(zendeskNote.channelType.getChannelId()),
-                notification = zendeskNote
-            )
-            return
-        }
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/push/NotificationMessageHandler.kt:113]: This removes the push notification logic to handle zendesk notifications, right?
Is this related to preventing submitting a request without a valid identity?
How do we plan to handle these notifications (not sure if they are sent) and how will we handle the communication, in general, when HE responds to a ticket?

Comment (wzieba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/push/NotificationMessageHandler.kt:113]: 👋 more context in this PR: #8521, (it was targeting this PR). We moved to emails, so notifications are not needed.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/support/requests/SupportRequestFormViewModelTest.kt [hunk 2]
```diff
@@ -164,14 +168,33 @@ internal class SupportRequestFormViewModelTest : BaseUnitTest() {
...
+        // When
+        sut.onHelpOptionSelected(TicketType.MobileApp)
+        sut.submitSupportRequest(mock(), HelpOrigin.LOGIN_HELP_NOTIFICATION, emptyList())
+
+        // Then
+        assertThat(event).hasSize(1)
+        assertThat(event[0]).isInstanceOf(ShowSupportIdentityInputDialog::class.java)
+        verify(tracks, times(1)).track(AnalyticsEvent.SUPPORT_NEW_REQUEST_FAILED)
+    }
+
```
Comment (atorresveiga) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/support/requests/SupportRequestFormViewModelTest.kt:197]: Nice!

**Review by atorresveiga — APPROVED** [review]
The code looks good and is working as expected, preventing sending the request without identity and tracking the correct events.

Evidence:



```
🔵 Tracked: support_new_request_failed, Properties: {"blog_id":216695598,"is_wpcom_store":true,"is_debug":true}
```

**Pull request merged**